src/viam/sdk/common/instance.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <memory>
+
+namespace viam {
+namespace sdk {
+
+/// @brief Instance management for Viam C++ SDK applications.
+/// This is a single instance class which is responsible for global setup and teardown related to
+/// the SDK. An Instance must be constructed before doing anything else in a program, and it must
+/// remain alive in a valid state for the duration of the program. Creating multiple Instance
+/// objects in the same program is an error.
+class Instance {
+   public:
+    Instance();
+    ~Instance();
+
+    static Instance& current();
```

Comment:
Do you want to doc-comment this, particularly if it is mostly intended for testing.